Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,207 @@
+  58% | assets/img  | Primarily hydrated by "Photoshop"
+
+Repository status:  Healthy
+Recommended action: None
+```
+
+## Background Information
+
+- **Modified paths** is the list of files which have been modified and thus correctly reflect everything on the disk. The paths here point to directories and files which have either been modified, created, or deleted. These files are handed to git when a git command reads its index in order to populate its list of files being tracked
+- **Placeholders** are files which have been opened by the local system with a read tag but without a write tag. They are objects tracked by VFS4G since git should not have to internally scan them since it is guaranteed that they have not been modified and accordingly do not have to be rehashed for differences. Their contents are on disk, but control over them has not been entirely ceded over to git. They decrease performance because VFS4G has to track all of them individually, and this can cause `git` commands that need to update the projection to run slowly
+- The **Git index** is git's internal list of files which it is tracking for creating  the next `commit`. It is useful to this project by means of comparing it to the list of **placeholders** to deduce if those files should truly be hydrated or if it may have occurred incidentally.
+- The **file system** must be carefully considered in the case of working with a mounted repository. VFS4G intercepts file I/O calls at the kernel level to inject its own behavior (the projected file system and if needed a network call to pull down the file which the system asked for in the correct format). Because of this, enumerating directories can cause additional data to be pulled down from the remote by hydrating extra files or further hydrating those that already exist on disk. Accordingly, care must be taken when interacting with the file system lest the tool designed to help users increase enlistment heath end up causing more problems.
+- There are several **files states** through which a file in the enlistment can transition. Depending on where files are in this life cycle, performance can be better or worse. Files begin as ***virtual files*** which do not exist on disk but are projected into the file system. They transition to ***empty placeholders*** when the file is opened without a write handle but data has yet to be read. Once data has been read from the file, it becomes a ***hydrated placeholder*** which is managed by VFS4G but has its entirety on the disk. If a file is opened with a write handle, it becomes a ***full file*** with control ceded to git. Full files are tracked in the modified paths list. Additionally, if a file is created it is a full file. The final file type is a ***tombstone***, a special hidden ProjFS file denoting that a file was deleted and thus it should not be projected into the system any longer.
+
+![File Life Cycle](FileLifeCycle.png)

```

Comment:
I think the diagram here is really nice.  

A few minor suggestions on the state transitions:

- Virtual Files always transition to Empty Placeholder first.  They will not transition directly to Tombstone or Full File (those two arrows can be removed).
- It might be worth mentioning in the diagram that "Virtual File" and "Tombstone" are Windows specific
- You could consider changing the "Data is read" text to something like "Data is required" as this transition will happen for reads or for writes that require the current file contents (e.g. appending to a file).